tests/cephadm/test_client.py

```diff
@@ -45,29 +45,36 @@ def put_file(client, file_name, content, file_mode, sudo=True):
         file_ = client.remote_file(sudo=sudo, file_name=file_name, file_mode=file_mode)
         file_.write(content)
         file_.flush()
+        file_.close()
 
     nodes_ = config.get("nodes", config.get("node"))
-    default_version = cls.cluster.rhcs_version.version[0]
+    default_version = str(cls.cluster.rhcs_version.version[0])
     use_cdn = cls.cluster.use_cdn
     if nodes_:
         if not isinstance(nodes_, list):
             nodes_ = [{nodes_: {}}]
 
         def setup(host):
-            rhcs_version = str(
-                [
-                    host[entry].get(
-                        "release", default_version if not use_cdn else "default"
-                    )
-                    for entry in host
-                ][0]
-            )
-            name = [entry for entry in host][0]
-            node = get_node_by_id(cls.cluster, name)
+            name = list(host.keys()).pop()
+            _build = list(host.values()).pop()
+            _node = get_node_by_id(cls.cluster, name)
+            if _build.get("release"):
+                rhcs_version = _build["release"]
+                if not isinstance(rhcs_version, str):
+                    rhcs_version = str(rhcs_version)
+
+                # Todo: Pointing to RHCS 5 CDN repo rpms, Since 6x not yet GAed
+                #       Revert this commit/code changes once 6x is GAed.
+                if rhcs_version > "5":
```

Comment:
This issue will come up once we start testing 7.x as well. Need to come up with another way to handle this automatically!

Reply:
That's right. As soon as 6x is released we can remove this for a while.